src/dsp_tools/commands/ingest_xmlupload/apply_ingest.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import glob
+from typing import cast
+
+import pandas as pd
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.user_information import IngestInformation
+from dsp_tools.models.exceptions import InputError
+
+
+def get_mapping_dict_from_file() -> dict[str, str]:
+    """
+    This functions returns the information to replace the original filepaths with the identifier from dsp-ingest.
+
+    Returns:
+        dictionary with original: identifier from dsp-ingest
+
+    Raises:
+        InputError: if no file was found
+    """
+    if filepath := glob.glob("mapping-*.csv"):
+        df = pd.read_csv(filepath[0])
+        print(f"The file '{filepath[0]}' is used to map the internal SIPI image IDs to the original filepaths.")
+        return dict(zip(df["original"].tolist(), df["derivative"].tolist()))
+    else:
+        raise InputError("No mapping csv file was found in the directory.")
+
+
+def replace_filepath_with_sipi_uuid(
+    xml_tree: etree._ElementTree[etree._Element],
+    orig_path_2_uuid_filename: dict[str, str],
+) -> tuple[etree._ElementTree[etree._Element], IngestInformation]:
+    """
+    Replace the original filepaths in the <bitstream> tags by the uuid filenames of the uploaded files.
+
+    Args:
+        xml_tree: The parsed original XML tree
+        orig_path_2_uuid_filename: Mapping from original filenames to uuid filenames from the mapping.csv
+
+    Returns:
+        The XML tree with the replaced filepaths (modified in place)
+        Message informing if all referenced files were uploaded or not.
+    """
+    no_uuid_found = []
+    used_media_paths = []
+    for elem in xml_tree.iter():
+        if etree.QName(elem).localname.endswith("bitstream"):
+            if (img_path := elem.text) in orig_path_2_uuid_filename:
+                elem.text = orig_path_2_uuid_filename[img_path]
+                used_media_paths.append(img_path)
+            else:
+                no_uuid_found.append((cast("etree._Element", elem.getparent()).attrib.get("id"), elem.text))
```

Comment:
yes, please! `from __future__` should go away soon-ish (not even sure it's needed in python 3.12), and it's generally better for tooling than the quoted variant

Reply:
To prevent misunderstandings: With "should go away soon-ish" you mean that it will soon become default behavior, so that it isn't necessary any more. You mean that quoting type annotations is not a good idea. So, you don't mean that `from __future__ import annotations` should be removed from this file here.

I'm not sure, but the last time I read something about type hints handling in python, they wrote that `from __future__ import annotations` won't disappear soon. I think we still need it now, and even in the middle term future.

Reply:
Correct, I mean that we should use `from __future__ import annotations` rather than quoted type annotations, because it will be easier to change, once the annotations are default behaviour.  
And you're right, this seems to be an ongoing thing - I just did a bit of reading on it... and things seem to move slowly - so for now we will just have to live with the future imports